src/main/java/com/singh/astha/medicinereminder/services/impl/MedicineServiceImpl.java

```diff
@@ -38,16 +42,19 @@ public class MedicineServiceImpl implements MedicineService {
     private final MedicineCategoryRepository medicineCategoryRepository;
     private final CategoryDtoTransformer categoryDtoTransformer;
 
+    private final DosageHistoryRepository dosageHistoryRepository;
+
     @Autowired
     public MedicineServiceImpl(MedicineRepository medicineRepository, MedicineDtoTransformer medicineDtoTransformer,
                                CategoryRepository categoryRepository,
                                MedicineCategoryRepository medicineCategoryRepository,
-                               CategoryDtoTransformer categoryDtoTransformer) {
+                               CategoryDtoTransformer categoryDtoTransformer, DosageService dosageService, DosageHistoryRepository dosageHistoryRepository) {
```

Comment:
I don't see any usage of `dosageService` here.

Also don't wire repository of some other domain here directtly. Instead create a DosageHelper and create a helper method for your  usecase.